bin/knc.c

```diff
@@ -795,7 +794,10 @@ move_local_to_network_buffer(work_t *work)
 		/* EOF */
 		return 0;
 	} else if (work->network_buffer.in_len < 0) {
-		if (errno == ECONNRESET)
+		const int errnum = errno;
+		if (errnum == EINTR || errnum == EAGAIN)
```

Comment:
And here check for `EWOULDBLOCK` per above comment.

Reply:
added check in subsequent commit